pkg/openshift/certgen/kubeconfig.go

```diff
@@ -267,7 +267,8 @@ func (c *Config) WriteMasterKubeConfigs(fs filesystem.Filesystem) error {
 		if err != nil {
 			return err
 		}
-		err = fs.WriteFile(filename, b, 0600)
+		fi := GetFileInfo(filename)
+		err = fs.WriteFile(filename, b, fi)
```

Comment:
your tests should have failed on this - you didn't specify 0600 for kubeconfigs in your rules so they'll be being written out as 0644.
1) update the tests
2) either update the rules or hardcode `filesystem.FileInfo{"root","root",0600}` here

Reply:
personally I'd continue to hardcode and not call GetFileInfo in all locations except in files.go

Reply:
Hard coded.